packages/amplify-graphql-api-construct/src/graphql-output.ts

```diff
@@ -0,0 +1,102 @@
+// Temp file until published in CLI
+
+import { CfnOutput, Stack } from 'aws-cdk-lib';
+import { z } from 'zod';
+
+const AwsAppsyncAuthenticationType = z.enum(['API_KEY', 'AWS_LAMBDA', 'AWS_IAM', 'OPENID_CONNECT', 'AMAZON_COGNITO_USER_POOLS']);
+export type AwsAppsyncAuthenticationType = z.infer<typeof AwsAppsyncAuthenticationType>;
+
+export const graphqlOutputSchemaV1 = z.object({
+  version: z.literal('1'),
+  payload: z.object({
+    awsAppsyncRegion: z.string(),
+    awsAppsyncApiEndpoint: z.string(),
+    awsAppsyncAuthenticationType: AwsAppsyncAuthenticationType,
+    awsAppsyncApiKey: z.string().optional(),
+  }),
+});
+
+export const versionedGraphqlOutputSchema = z.discriminatedUnion('version', [
+  graphqlOutputSchemaV1,
+  // this is where additional api major version schemas would go
+]);
+
+export type GraphqlOutput = z.infer<typeof versionedGraphqlOutputSchema>;
+
+export const GraphqlOutputKey = 'graphqlOutput';
+
+// Types that need to be exported from CLI types package
+type BackendOutputEntry<T extends Record<string, string> = Record<string, string>> = {
+  readonly version: string;
+  readonly payload: T;
+};
+
+type BackendOutput = Record<string, BackendOutputEntry>;
+
+export type BackendOutputStorageStrategy<T extends BackendOutputEntry> = {
+  addBackendOutputEntry(keyName: string, backendOutputEntry: T): void;
+
+  /**
+   * Write all pending data to the destination
+   */
+  flush(): void;
+};
+
+export const amplifyStackMetadataKey = 'AWS::Amplify::Output';
```

Comment:
We should probably use a different metadata key. `AWS::Amplify::Output::Graphql` or something along those lines.

We would also need the CLI to make this configurable in the exported version of `StackMetadataBackendOutputStorageStrategy`.

Reply:
It'd need to be the same to have `generate` work consistently, right?

Reply:
Oh yes, you are right. One complication, I think we can only flush once for any given key. Customers would not be able to have multiple constructs with the manual config.

Reply:
Can we read/mutate the key?